File: platforms/ios/foundation/Sources/NativeblocksFoundation/modifier/Alpha.swift
import NativeblocksCompiler
import NativeblocksRuntime
import SwiftUI

@Modifier(
    name: "Alpha",
    keyType: "nativeblocks/alpha",
    description: "Sets the opacity of the block.",
    version: 1,
    versionName: "1"
)
struct Alpha: ViewModifier {

    @ModifierData(description: "Opacity between 0.0 (transparent) and 1.0 (opaque).", defaultValue: "1.0")
    var value: Double = 1

    func body(content: Content) -> some View {
        content.opacity(min(max(value, 0), 1))
    }
}

File: platforms/android/foundation/foundation/src/main/java/io/nativeblocks/foundation/modifier/Alpha.kt
package io.nativeblocks.foundation.modifier

import androidx.compose.ui.draw.alpha
import io.nativeblocks.compiler.type.Modifier
import io.nativeblocks.compiler.type.ModifierData
import androidx.compose.ui.Modifier as UiModifier

@Modifier(
    keyType = "nativeblocks/alpha",
    name = "Alpha",
    description = "Sets the opacity of the block.",
    version = 1,
    versionName = "1",
)
internal fun alpha(
    @ModifierData(description = "Opacity between 0.0 (transparent) and 1.0 (opaque).", defaultValue = "1.0")
    value: Float = 1f,
): UiModifier = UiModifier.alpha(value.coerceIn(0f, 1f))
